cub/cub/device/device_segmented_radix_sort.cuh

```diff
@@ -182,7 +182,10 @@ struct DeviceSegmentedRadixSort
    *   sequence of associated value items
    *
    * @param[in] num_items 
-   *   The total number of items to sort (across all segments)
+   *   The total number of items to sort (across all segments). If segments
+   *   are non-contiguous, this should include unused items in between
+   *   segments, namely all items between the start of the first and end
+   *   of the last segment.
```

Comment:
I believe this definition would miss items before the first segment and after the last one. Do you think something along the following lines would be better? 
```suggestion
   *   The total number of items within the segmented array, including items not covered by
   *   segments. If there's a segment that covers the last element of the array, `num_items` should 
   *   match the largest element within the range `[d_end_offsets, d_end_offsets + num_segments)`.
```

Reply:
After taking a closer look with @elstehle, we think that the `If there's a segment that covers the last element of the array, ` can be omitted. We have to know about the first uncovered items, but the last ones should not matter.

Reply:
That's a good point, yes I think your suggestion makes sense.